include/swift/AST/Decl.h

```diff
@@ -3841,7 +3842,8 @@ class ClassDecl final : public NominalTypeDecl {
   ///
   /// \see getForeignClassKind
   bool isForeign() const {
-    return getForeignClassKind() != ForeignKind::Normal;
+    return getForeignClassKind() != ForeignKind::Normal ||
+      const_cast<ClassDecl *>(this)->isForeignReferenceType();
```

Comment:
This is not ideal, but I think it's OK because we never construct a const decl, right?